source/replication/enable-server-side-two-way-bucket-replication.rst

```diff
@@ -0,0 +1,514 @@
+procedure to ensure consistent object replication.
+
+Replication of Delete Operations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+MinIO supports replicating delete operations onto the target bucket. 
+Specifically, MinIO can replicate versioning
+:s3-docs:`Delete Markers <versioning-workflows.html>` and the deletion
+of specific versioned objects:
+
+- For delete operations on an object, MinIO replication also creates the delete
+  marker on the target bucket.
+
+- For delete operations on versions of an object,
+  MinIO replication also deletes those versions on the target bucket.
+
+MinIO requires explicitly enabling replication of delete operations using the
+:mc-cmd-option:`mc replicate add replicate` or 
+:mc-cmd-option:`mc replicate edit replicate`. This procedure includes the
+required flags for enabling replication of delete operations and delete markers.
+
+Replication of Encrypted Objects
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+MinIO supports replicating objects encrypted with automatic 
+Server-Side Encryption (SSE-S3). Both the source and destination buckets
+*must* have automatic SSE-S3 enabled for MinIO to replicate an encrypted object.
+
+As part of the replication process, MinIO *decrypts* the object on the source
+bucket and transmits the unencrypted object. The destination MinIO cluster then
+re-encrypts the object using the destination bucket SSE-S3 configuration. MinIO
+*strongly recommends* :ref:`enabling TLS <minio-TLS>` on both source and
+destination clusters to ensure the safety of objects during transmission.
+
+MinIO does *not* support replicating client-side encrypted objects 
+(SSE-C).
+
+Procedure
+---------
+
+1) Configure User Accounts and Policies for Replication
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This step creates multiple users and policies on both MinIO clusters for
+supporting replication operations. You can skip this step if both
+clusters already have users with the necessary
+:ref:`permissions <minio-bucket-replication-serverside-twoway-permissions>`.
+
+The following examples that the :mc:`alias <mc alias>` for each cluster
+provides the necessary permissions for creating policies and users on both
+clusters. See :ref:`minio-users` and :ref:`minio-policy` for more complete
+documentation on MinIO users and policies respectively.
+
+Replace ``Alpha`` and ``Baker`` in the following examples with the appropriate
+:mc:`aliases <mc alias>` for the MinIO clusters on which you are configuring
+bucket replication.
+
+A\) Create Replication Administrators
+   The following code creates policies and users for supporting configuring
+   replication on the ``Alpha`` and ``Baker`` clusters. Replace the
+   password ``LongRandomSecretKey`` with a long, random, and secure secret key 
+   as per your organizations best practices for password generation.
+
+   .. code-block:: shell
+      :class: copyable
+
+      wget https://docs.min.io/minio/baremetal/examples/ReplicationAdminPolicy.json | \
```

Comment:
Note: These URLS won't resolve until this is public. In the short-term you can wget the files from `source/examples/`